### ISSUE TITLE
Downgrade to logback 1.1.6 (or upgrade to 1.1.8 when available)

### ISSUE THREAD

**Issue by mdeinum** [spring-projects/spring-boot#6214]
While upgrading to Spring Boot 1.3.5 we ran into [LOGBACK-1158](http://jira.qos.ch/browse/LOGBACK-1158). The suggested solution is to downgrade, we now done that in our `gradle.build` file. 

Didn't took too long to fix but it was quite unexpected that this broke our application.

**Comment by philwebb** [comment]
I don't think we should downgrade due to a logback issue, especially as it's in the `SMTPAppender` so it won't affect all users. We'll pickup 1.1.8 in due course.

**Comment by sbuettner** [comment]
We also have problem with logstash `1.1.6` since we are using the `logstash-logback-encoder`: https://github.com/logstash/logstash-logback-encoder/issues/160 which seems to be caused by http://jira.qos.ch/browse/LOGBACK-1164

Due to that we currently have to downgrade every project explicitly to `1.1.6`.

**Comment by jltnf54** [comment]
just ran into this issue too, had to downgrade to 1.1.6

**Comment by ptzhuf** [comment]
`   <appender name="logstash" class="net.logstash.logback.appender.LogstashTcpSocketAppender">
        <destination>192.168.153.121:4560</destination>
        
        <encoder class="net.logstash.logback.encoder.LogstashEncoder"/>
        
    </appender>`
`Exception in thread "main" java.lang.IllegalStateException: Logback configuration error detected: 
ERROR in ch.qos.logback.core.joran.action.NestedBasicPropertyIA - Unexpected aggregationType AS_BASIC_PROPERTY_COLLECTION
	at org.springframework.boot.logging.logback.LogbackLoggingSystem.loadConfiguration(LogbackLoggingSystem.java:152)
	at org.springframework.boot.logging.logback.LogbackLoggingSystem.reinitialize(LogbackLoggingSystem.java:195)`

**just ran into this issue too, had to downgrade to 1.1.6**

ch.qos.logback:logback-classic:1.1.7
org.springframework.boot:spring-boot-starter-web:1.4.0.RELEASE
org.springframework.boot:spring-boot-starter-actuator:1.4.0.RELEASE

**Comment by ramblingred** [comment]
Same issue here. My team actually decided not to upgrade our services to SB 1.4 yet instead of overriding logback across all of our microservices.  Not ideal, but hopefully 1.1.8 comes out soon and can be integrated into SB quickly.

**Comment by jonashackt** [comment]
@ramblingred same for us - still at 1.3.x, longing for 1.4

**Comment by aburmeis** [comment]
Has been fixed in Logback Core 30th of March (https://github.com/qos-ch/logback/commit/2ba396a20e44d2f0985987ddc68ba410f24078ae) but no release since then...

**Comment by SalmanAwan** [comment]
Ran into same issue when upgraded SpringBoot from 1.3.3 to 1.4.1 on a gradle 2.2.1 multi-module project.

Below suggested fix didn't work for me:


Following temp fix (downgrading to logback 1.1.6) worked for me. 

configure (services) {
        configurations {
                .....
                all*.exclude module: 'logback-classic'
        }
        dependencies {
                .....
                compile 'ch.qos.logback:logback-classic:1.1.6'
        }
}

**Comment by SalmanAwan** [comment]
It might help some, from [Gradle Dependency Management](https://docs.gradle.org/current/userguide/dependency_management.html)

Below temp fix (till logback 1.1.8 is out) worked for my multi-module gradle project when upgrading SpringBoot to 1.4.1. (You might want to remove `allprojects { ... }` for single module gradle projects)

```
allprojects {
    configurations.all {
        resolutionStrategy.eachDependency { DependencyResolveDetails details ->
            if (details.requested.name == 'logback-classic' || details.requested.name == 'logback-core') {
                details.useVersion '1.1.6'
}}}}
```

**Comment by ianhattendorf** [comment]
1.1.8 is out as of December 9th

http://jira.qos.ch/projects/LOGBACK/versions/11320
